src/App.tsx

```diff
@@ -1,61 +1,144 @@
 import './App.scss';
+import { FC, useState, useEffect } from 'react';
+import usersFromServer from './api/users';
+import todosFromServer from './api/todos';
+// eslint-disable-next-line import/no-cycle
+import { TodoList } from './components/TodoList';
 
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+export interface Todo {
+  id: number,
+  title: string,
+  completed: boolean,
+  userId: number,
+  user?: User | null,
+}
+
+export interface User {
+  id: number,
+  name: string,
+  username: string,
+  email: string
+}
+
+export const App: FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [users, setUsers] = useState<User[]>([]);
+
+  const [newTodoTitle, setNewTodoTitle] = useState('');
+  const [newUserId, setNewUserId] = useState<number | null>(null);
+
+  const [titleErrorMessage, setTitleErrorMessage] = useState('');
+  const [userErrorMessage, setUserErrorMessage] = useState('');
+
+  useEffect(() => {
+    setUsers(usersFromServer);
+    setTodos(todosFromServer);
+  }, []);
+
+  function getUser(userId: number): User | null {
+    const foundUser = users.find(user => user.id === userId);
+
+    return foundUser || null;
+  }
+
+  const currentTodos: Todo[] = todos.map(todo => ({
+    ...todo,
+    user: getUser(todo.userId),
+  }));
```

Comment:
Let's move this logic outside our `App` component to avoid re-creating every render

```suggestion
  const currentTodos: Todo[] = usersFromServer.map(todo => ({
    ...todo,
    user: getUser(todo.userId),
  }));
```